### PR TITLE
Added docs for pending discord alert handler

### PR DESCRIPTION
**Pull request by mattnotmitt**
Documentation for https://github.com/influxdata/kapacitor/pull/2287/. Mostly copied and corrected from the Slack config as I used it for the base of the kapacitor implementation.

### CODE REVIEW

**Comment by sanderson** [comment]
This docs PR is dependent on this Kapacitor PR (https://github.com/influxdata/kapacitor/pull/2287) and shouldn't be merged until the associated PR is merged and a new version of Kapacitor is released.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
+---
+title: Discord event handler
+description: The Discord event handler allows you to send Kapacitor alerts to Discord. This page includes configuration options and usage examples.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: allows you to >> lets you

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+menu:
+  kapacitor_1_5:
+    name: Discord
+    weight: 250
+    parent: event-handlers
+---
+
+[Discord](https://discordapp.com) is a popular chat service targeted primarily at gamers but is used by some teams.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: but is used >> and used....by teams outside of gaming?

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+Kapacitor can be configured to send alert messages to Discord.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: To configure Kapacitor to send..., set the applicable configuration options.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+## Configuration
+Configuration as well as default [option](#options) values for the Discord event
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set configuration and default [options] values for the Discord event handler in your `kapacitor.conf` file. See example configuration below:

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+```
+
+> Multiple Discord clients may be configured by repeating `[[discord]]` sections.
+The `workspace` acts as a unique identifier for each configured Discord client.
+
+#### `enabled`
+Set to `true` to enable the Discord event handler.
+
+#### `default`
+Identify one of the Discord configurations as the default if there are multiple
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: For multiple Discord client configurations, identify one configuration as the default.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+Discord configurations.
+
+#### `workspace`
+The Discord workspace ID.
+This can be any string that identifies this particular Discord configuration.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set the string to identify the Discord configuration.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+A logical choice is the name of the Discord channel and the guild it's a part
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: For example, the name...part of, such as `<guild>-<channel>`.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+A logical choice is the name of the Discord channel and the guild it's a part 
+of, e.g. `<guild>-<channel>`.
+
+#### `timestamp`
+Boolean signifying whether the timestamp should be shown in the embed.
+
+#### `url`
+The Discord webhook URL. This can be obtained by adding an Incoming Webhook integration.
+Follow [this guide](https://support.discordapp.com/hc/en-us/articles/228383668) and add a webhook for Kapacitor.
+Discord will provide you with the webhook URL.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Obtain by adding...
To obtain a webhook URL from Discord, see [Intro to Webhooks](...)

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+
+#### `username`
+The username of the Discord bot. If set, this will override the username
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set the Discord bot username to override...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+The Discord webhook URL. This can be obtained by adding an Incoming Webhook integration.
+Follow [this guide](https://support.discordapp.com/hc/en-us/articles/228383668) and add a webhook for Kapacitor.
+Discord will provide you with the webhook URL.
+
+#### `username`
+The username of the Discord bot. If set, this will override the username 
+set when generating the webhook.
+
+#### `avatar-url`
+A URL pointing to the desired avatar of the bot. If set, this will
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set a URL to a specified avatar to override the....

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+override the avatar set when generating the webhook.
+
+#### `embed-title`
+The title displayed in the alert embed. If left blank, no title will
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set the title to display... If blank, no title is set.

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+be set
+
+#### `global`
+If true all the alerts will be sent to Discord without explicitly specifying Discord
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set to `true` to send all alerts to Discord...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+in the TICKscript.
+
+#### `state-changes-only`
+Sets all alerts in state-changes-only mode, meaning alerts will only be sent if
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set to `true` to send all alerts in state-changes-only mode (alert sent if alert state changes).

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+_Only applies if `global` is `true`._
+
+#### `ssl-ca`
+Path to certificate authority file.
+
+#### `ssl-cert`
+Path to host certificate file.
+
+#### `ssl-key`
+Path to certificate private key file.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set path

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+#### `state-changes-only`
+Sets all alerts in state-changes-only mode, meaning alerts will only be sent if
+the alert state changes.
+_Only applies if `global` is `true`._
+
+#### `ssl-ca`
+Path to certificate authority file.
+
+#### `ssl-cert`
+Path to host certificate file.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set path

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+If true all the alerts will be sent to Discord without explicitly specifying Discord
+in the TICKscript.
+
+#### `state-changes-only`
+Sets all alerts in state-changes-only mode, meaning alerts will only be sent if
+the alert state changes.
+_Only applies if `global` is `true`._
+
+#### `ssl-ca`
+Path to certificate authority file.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set path...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+
+#### `ssl-key`
+Path to certificate private key file.
+
+#### `insecure-skip-verify`
+Use SSL but skip chain and host verification.
+_This is necessary if using a self-signed certificate._
+
+## Options
+The following Discord event handler options can be set in a
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set the following Discord....options in...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+Path to certificate authority file.
+
+#### `ssl-cert`
+Path to host certificate file.
+
+#### `ssl-key`
+Path to certificate private key file.
+
+#### `insecure-skip-verify`
+Use SSL but skip chain and host verification.
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set to `true` to use...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+  // ...
+  .discord()
+    .workspace('guild-channel')
+    .username('Kapacitor')
+    .avatarUrl('https://influxdata.github.io/branding/img/downloads/influxdata-logo--symbol--pool-alpha.png')
+    .timestamp(true)
+    .embedTitle('Kapacitor Alert')
+```
+
+## Guild Setup
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Set up Guild

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+To allow Kapacitor to send alerts to Discord, follow [this guide](https://support.discordapp.com/hc/en-us/articles/228383668)
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: To allow...., obtain a webhook URL from Discord, see [Intro to Webhooks](...). Then, add the generated webhook URL...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 21]
```diff
@@ -0,0 +1,314 @@
...
+With one or more Discord event handlers enabled and configured in your
+`kapacitor.conf`, use the `.discord()` attribute in your TICKscripts to send
+alerts to Discord or define a Discord handler that subscribes to a topic and sends
+published alerts to Discord.
+
+> To avoid posting a message every alert interval, use
+> [AlertNode.StateChangesOnly](/kapacitor/v1.5/nodes/alert_node/#statechangesonly)
+> so only events where the alert changed state are sent to Discord.
+
+The examples below use the following Discord configurations defined in the `kapacitor.conf`:
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: See examples below for sample Discord config...

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+  timestamp = true
+  username = "StatsBot"
+  avatar-url = "https://influxdata.github.io/branding/img/downloads/influxdata-logo--symbol--pool-alpha.png"
+  embed-title = "Errors"
+  global = false
+  state-changes-only = false
+```
+
+### Send alerts to Discord from a TICKscript
+The following TICKscript uses the `.discord()` event handler to send the message,
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Use the `.discord()`...in your TICKscript to send an alert. For example:

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 23]
```diff
@@ -0,0 +1,314 @@
...
+    .warn(lambda: "usage_idle" < 20)
+    .stateChangesOnly()
+    .message('Hey, check your CPU')
+    .discord()   
+      .embedTitle('Uh Oh!')  
+```
+
+### Send alerts to Discord from a defined handler
+
+The following setup sends an alert to the `cpu` topic with the message,
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Add a Discord handler that subscribes to....by creating a TICKscript that publishes alert messages to a topic. For example:

File: content/kapacitor/v1.5/event_handlers/discord.md [hunk 24]
```diff
@@ -0,0 +1,314 @@
...
+  |alert()
+    .crit(lambda: "usage_idle" < 5)
+    .stateChangesOnly()
+    .message('Hey, I think the machine is on fire.')
+    .discord()
+      .workspace('alerts')
+      .embedTitle('AAAAAAAAAAAAAAAAAAAAAA')
+```
+
+Error rates are also being stored in the same InfluxDB instance and we want to
```
Comment (kelseiv) [inline content/kapacitor/v1.5/event_handlers/discord.md]: Error rates are also stored in...

**Comment by kelseiv** [comment]
Thank you, @mattnotmitt! Minor grammatical edits.

**Comment by mattnotmitt** [comment]
Hey @kelseiv thanks for the guidance, have made the requested edits now! Hopefully the Kap PR will be merged soon so this can be merged too.

**Comment by mattnotmitt** [comment]
Looks like this is targeted to be released in Kapacitor 1.5.6.

**Pull request merged**